auto_dev/commands/run.py

```diff
@@ -29,19 +29,32 @@
 class AgentRunner:
     """Class to manage running an agent."""
 
-    agent_name: str
+    agent_name: PublicId
     verbose: bool
     force: bool
     logger: Any
+    no_fetch: bool = False
 
     def run(self) -> None:
         """Run the agent."""
-
-        self.logger.info(f"Fetching agent {self.agent_name} from the local package registry...")
+        
+        if self.no_fetch:
+            self.logger.info(f"Looking for local agent package in directory {self.agent_name.name}...")
+            if not Path(self.agent_name.name).exists():
+                self.logger.error(f"Local agent package {self.agent_name.name} does not exist.")
+                sys.exit(1)
+            self.logger.info(f"Found local agent package at {self.agent_name.name}")
+            self.logger.info(f"Changing to directory: {self.agent_name.name}")
+            self.change_directory(self.agent_name.name)
+        else:
+            self.logger.info(f"Fetching agent {self.agent_name} from the local package registry...")
+            if self.check_agent_exists():
+                sys.exit(1)
+            self.fetch_agent()
+            self.logger.info(f"Changing to directory: {self.agent_name.name}")
```

Comment:
this is quite repetitive, can we reverse the logic, call it self.fetch for one, check if its set to fetch and do that.
Can even only do one call for the check if the agent exissts.